src/contracts/core/PaymentCoordinator.sol

```diff
@@ -0,0 +1,501 @@
+        TokenTreeMerkleLeaf calldata tokenLeaf
+    ) internal pure {
+        // Validate index size so that there aren't multiple valid indices for the given proof
+        // index can't be greater than 2**(tokenProof/32)
+        require(
+            tokenLeafIndex < (1 << (tokenProof.length / 32)),
+            "PaymentCoordinator._verifyTokenClaim: invalid tokenLeafIndex"
+        );
+
+        // Verify inclusion of token leaf
+        bytes32 tokenLeafHash = calculateTokenLeafHash(tokenLeaf);
+        require(
+            Merkle.verifyInclusionKeccak({
+                root: earnerTokenRoot,
+                index: tokenLeafIndex,
+                proof: tokenProof,
+                leaf: tokenLeafHash
+            }),
+            "PaymentCoordinator._verifyTokenClaim: invalid token claim proof"
+        );
+    }
+
+    /**
+     * @notice verify inclusion of earner claim proof in the distribution root. This verifies
+     * the inclusion of the earner and earnerTokenRoot hash in the tree. The token claims are proven separately
+     * against the earnerTokenRoot hash (see _verifyTokenClaimProof). The earner leaf comprises of (earner, earnerTokenRoot)
+     * @param root distribution root that should be read from storage
+     * @param earnerLeafIndex index of the earner leaf
+     * @param earnerProof proof of the earners account root in the merkle tree
+     * @param earnerLeaf leaf of earner merkle tree containing the earner address and earner's token root hash
+     */
+    function _verifyEarnerClaimProof(
+        bytes32 root,
+        uint32 earnerLeafIndex,
+        bytes calldata earnerProof,
+        EarnerTreeMerkleLeaf calldata earnerLeaf
+    ) internal pure {
+        // Validate index size so that there aren't multiple valid indices for the given proof
+        // index can't be greater than 2**(earnerProof/32)
+        require(
+            earnerLeafIndex < (1 << (earnerProof.length / 32)),
+            "PaymentCoordinator._verifyEarnerClaimProof: invalid earnerLeafIndex"
+        );
+        // Verify inclusion of earner leaf
+        bytes32 earnerLeafHash = calculateEarnerLeafHash(earnerLeaf);
+        require(
+            Merkle.verifyInclusionKeccak({
+                root: root,
+                index: earnerLeafIndex,
+                proof: earnerProof,
+                leaf: earnerLeafHash
+            }),
+            "PaymentCoordinator._verifyEarnerClaimProof: invalid earner claim proof"
+        );
+    }
+
+    /**
+     * @notice Sets the address of the entity that can claim payments on behalf of the earner
+     * @param claimer The address of the entity that can claim payments on behalf of the earner
+     * @dev Only callable by the `earner`
+     */
+    function setClaimerFor(address claimer) external {
+        address earner = msg.sender;
+        address prevClaimer = claimerFor[earner];
+        claimerFor[earner] = claimer;
+        emit ClaimerForSet(earner, prevClaimer, claimer);
+    }
```

Comment:
This and other state changing `external` methods should be above our `internal` methods, just to keep with the convention used in other files.

`pure` and `view` methods can stay down here

Reply:
separated by external, internal, and then view/pure
https://github.com/Layr-Labs/eigenlayer-contracts/pull/514/commits/02a4fcacbb2016a91bfe09150ddc758721d26540#diff-2a28246efa54c09dff5f0afef55f490d130b1a13c5b7e30efb51d5647c52a1a2